packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+    await container.generate(1)
+
+    // list ICX orders
+    const ordersAfterCreateOrder: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
+    expect((ordersAfterCreateOrder as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        status: ICXOrderStatus.OPEN,
+        type: ICXOrderType.INTERNAL,
+        tokenFrom: symbolDFI,
+        chainTo: order.chainTo,
+        receivePubkey: order.receivePubkey,
+        ownerAddress: order.ownerAddress,
+        amountFrom: order.amountFrom,
+        amountToFill: order.amountFrom,
+        orderPrice: order.orderPrice,
+        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
+        height: expect.any(BigNumber),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
+
+    const accountBTCBeforeOffer: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+    // make offer with invalid ownerAddress "123" - taker
+    const offer: ICXOffer = {
+      orderTx: createOrderTxId,
+      amount: new BigNumber(0.1), // 10 DFI = 0.1 BTC
+      ownerAddress: 'INVALID_OWNER_ADDRESS'
+    }
+
+    const promise = client.icxorderbook.makeOffer(offer, [])
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('RpcApiError: \'recipient (INVALID_OWNER_ADDRESS) does not refer to any valid address\', code: -5, method: icx_makeoffer')
+
+    const accountBTCAfterOffer: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+    // check same balance
+    expect(accountBTCAfterOffer[idDFI]).toStrictEqual(accountBTCBeforeOffer[idDFI])
+  })
+
+  it('should return an error with invalid ICXOffer.receivePubkey', async () => {
+    // create order - maker
+    const order: ICXOrder = {
+      chainFrom: 'BTC',
+      tokenTo: idDFI,
+      ownerAddress: accountDFI,
+      amountFrom: new BigNumber(2),
+      orderPrice: new BigNumber(100)
+    }
+    const createOrderResult: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = createOrderResult.txid
+    await container.generate(1)
+
+    // list ICX orders
+    const ordersAfterCreateOrder: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
+    expect((ordersAfterCreateOrder as Record<string, ICXOrderInfo>)[createOrderTxId]).toStrictEqual(
+      {
+        status: ICXOrderStatus.OPEN,
+        type: ICXOrderType.EXTERNAL,
+        tokenTo: symbolDFI,
+        chainFrom: order.chainFrom,
+        ownerAddress: order.ownerAddress,
+        amountFrom: order.amountFrom,
+        amountToFill: order.amountFrom,
+        orderPrice: order.orderPrice,
+        amountToFillInToAsset: order.amountFrom.multipliedBy(order.orderPrice),
+        height: expect.any(BigNumber),
+        expireHeight: expect.any(BigNumber)
+      }
+    )
```

Comment:
you might not want to assert these in the non `icx_listorders` test, might be too overwhelming for you to assert these in this testing scope (icx_makerofer). It's up to you, IMO we can just d a similar check of:

```suggestion
    expect((ordersAfterCreateOrder[createOrderTxId].status).toStrictEqual(ICXOrderStatus.OPEN)
```

then within `icx_listorders` you test all the various states there.

Reply:
yes. I would not do that if I had to do it one by one. If you check, there are only a few different templates. so copy and paste and some run though. 
But I wonder if removing these extra asserts will improve test readability ??

Reply:
It's more to do with assertion readability because every assertion = "call to attention" of things you are validating/asserting, you had many assertions in your tests. This increases code review time for me as I need to see what you are asserting and compare it to your described scenario.

Reply:
`icx_listorders` might be a better place to do all the assertion of orders in detail, while for other calls we can assert basic soundness?

Reply:
agree. will refactor accordingly in the comming PRs